lib/src/embedded/message-transformer.ts

```diff
@@ -83,25 +81,11 @@ export class MessageTransformer {
     private readonly outboundProtobufs$: Observable<Buffer>,
     private readonly writeInboundProtobuf: (buffer: Buffer) => void
   ) {
-    this.outboundProtobufs$.subscribe(
-      buffer => {
-        try {
-          this.outboundMessagesInternal$.next(decode(buffer));
-        } catch (error) {
-          this.error$.next(error);
-        }
-      },
-      error => this.error$.next(error),
-      () => {
-        this.outboundMessagesInternal$.complete();
-        this.error$.complete();
-      }
+    this.outboundProtobufs$.pipe(map(buffer => decode(buffer))).subscribe(
```

Comment:
```suggestion
    this.outboundProtobufs$.pipe(map(decode)).subscribe(
```

Since `decode` is a global function, it's safe to pass in directly to a function. It's only instance methods that need to be wrapped in closures so they preserve their `this` context.